pkg/hive/hive.go

```diff
@@ -88,7 +88,7 @@ func (s *Service) sendPeers(ctx context.Context, peer swarm.Address, peers []swa
 	if err != nil {
 		return fmt.Errorf("new stream: %w", err)
 	}
-	defer stream.Close()
+	defer stream.FullClose()
```

Comment:
this will block and cause other components to block until this is done. i am not sure if `defer stream.FullClose()` or `defer func(){ go stream.FullClose() }()`

Reply:
Yes, but it was already like this. The only difference that now on error the stream is reset, so it will not block as it will already be reset. If you look at the end of the function, there use to be FullClose call if there was no error, so the functionality should be the same, but more correct now cuz the stream will be reset on error.

Reply:
Actaully, I would change it to return stream.FullClose as it was

Reply:
Oooh, sorry, I remember now. since we are sending in batches, I wanted to get the response that the stream is closed before opening a new one so bathes have a bit more sense.